sale_timesheet_enhanced/README.rst

```diff
@@ -0,0 +1,139 @@
+
+.. |badge1| image:: https://img.shields.io/badge/maturity-Beta-yellow.png
+    :target: https://odoo-community.org/page/development-status
+    :alt: Beta
+.. |badge2| image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fsale--workflow-lightgray.png?logo=github
+    :target: https://github.com/OCA/sale-workflow/tree/12.0/sale_timesheet_enhanced
+    :alt: OCA/sale-workflow
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_timesheet_enhanced
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/167/12.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module enhances selling timesheets in your sales order.
+
+End-User Enhancements:
+ * Usability improvements of Project > Edit > Invoicing
+ * Update Sale Order Line action on Timesheet entry
+ * Usability improvements of Task related to invoicing
+ * Exclude specific Task from Sale Order
+ * Usability improvements of Task > Timesheets
+ * Exclude specific Timesheet line from Sale Order
+
+Technical Enhancements:
+ * `account.analytic.line` extendability improvements:
+   * `_timesheet_get_sale_line` overridable
+ * `project.create.sale.order` extendability improvements:
+   * `action_create_sale_order` refactor
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+Excluding Timesheet line from Sale Order
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This feature proves itself useful for *By Task* billing approach, when
+a specific timesheet entry tracked towards a billable task needs to be
+excluded from the Sale Order.
+
+To exclude a specific timesheet entry from Sale Order:
+
+# Go to *Timesheets > Timesheet > All Timesheets*
+# Check *Exclude From Billing* for a specific timesheet entry
+
+or:
+
+# Go to *Project > All Tasks*
+# Open specific task for editing
+# Switch to *Timesheets* tab
+# Check *Exclude From Billing* for a specific timesheet entry
+
+Excluding Task from Sale Order
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This feature proves itself useful for *By Employee* billing approach, when
+a any timesheet entry tracked towards a specific task need to be excluded
+from the Sale Order.
+
+# Go to *Project > All Tasks*
+# Open specific task for editing
+# Check *Exclude From Billing*
+
+After adjusting Task billing settings, it may be needed to update the Sale
+Order Line of corresponding Timesheet entries. In order to do so:
+
+# Go to *Timesheets > Timesheet > All Timesheets*
+# Filter entries by task and select all entries logged towards altered Task
+# Execute *Action > Update Sale Order Line* action
+
```

Comment:
Final users could (will) forget making this extra-task with the action. This should be done automatically without any action.